Compilation Errors led by flow offload code of mediatek.

反馈bug/问题模板，提建议请删除

## 1.关于你要提交的问题

Q：是否搜索了issue (使用 "x" 选择)
* [x] 没有类似的issue

## 2. 详细叙述
No more details.
### (1) 具体问题

A：Multiple Definition in codes of mtwifi, which makes the compilation errors happen.

### (2) 路由器型号和固件版本

A：Xiaomi Redmi Router AC2100

### (3) 详细日志

A：

```
cp -fpR /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/root-ramips /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/root.orig-ramips
chmod: cannot access '/sbin/mtkwifi': No such file or directory
/home/chitanda/lede/build_dir/target-mipsel_24kc_musl/root-ramips/etc/init.d/nginx: line 10: /usr/bin/nginx-util: No such file or directory
Enabling autoreboot
Enabling boot
Enabling bootcount
Enabling cron
Enabling ddns
Enabling dnsmasq
Enabling done
Enabling dropbear
Enabling firewall
chmod: cannot access '/sbin/mtkwifi': No such file or directory
Enabling firmware.sh
Enabling fstab
Enabling gpio_switch
Enabling led
Enabling log
Enabling network
./etc/init.d/nginx: line 10: /usr/bin/nginx-util: No such file or directory
Enabling nginx
Enabling nlbwmon
Enabling rpcd
Enabling set-irq-affinity
Enabling sfe
Enabling shadowsocksr
Enabling shortcut-fe
Enabling sysctl
Enabling sysfixtime
Enabling sysntpd
Enabling system
Enabling uhttpd
Enabling umount
Enabling unblockmusic
Enabling urandom_seed
Enabling urngd
Enabling uwsgi
Enabling zram
sed -i "s/Installed-Time: .*/Installed-Time: 1603263141/" /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/root-ramips/usr/lib/opkg/status
rm -rf /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/root-ramips/tmp/*
rm -f /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/root-ramips/usr/lib/opkg/lists/*
rm -f /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/root-ramips/usr/lib/opkg/info/*.postinst*
rm -f /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/root-ramips/var/lock/*.lock
rm -rf /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/root-ramips/boot
find /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/root-ramips/ -mindepth 1 -execdir touch -hcd "@1603263141" "{}" +
make[2]: Leaving directory '/home/chitanda/lede'
make[2]: Entering directory '/home/chitanda/lede'
make[3]: Entering directory '/home/chitanda/lede/target/linux'
make[4]: Entering directory '/home/chitanda/lede/target/linux/ramips'
rm -f /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/localversion
/home/chitanda/lede/scripts/kconfig.pl  + /home/chitanda/lede/target/linux/generic/config-5.4 /home/chitanda/lede/target/linux/ramips/mt7621/config-5.4 > /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.target
awk '/^(#[[:space:]]+)?CONFIG_KERNEL/{sub("CONFIG_KERNEL_","CONFIG_");print}' /home/chitanda/lede/.config >> /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.target
echo "# CONFIG_KALLSYMS_EXTRA_PASS is not set" >> /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.target
echo "# CONFIG_KALLSYMS_ALL is not set" >> /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.target
echo "CONFIG_KALLSYMS_UNCOMPRESSED=y" >> /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.target
/home/chitanda/lede/scripts/package-metadata.pl kconfig /home/chitanda/lede/tmp/.packageinfo /home/chitanda/lede/.config 5.4 > /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.override
/home/chitanda/lede/scripts/kconfig.pl 'm+' '+' /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.target /dev/null /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.override > /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.set
mv /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.set /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.old
grep -v INITRAMFS /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.old > /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.set
echo 'CONFIG_INITRAMFS_SOURCE=""' >> /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.set
echo '# CONFIG_INITRAMFS_FORCE is not set' >> /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.set
rm -rf /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/modules
cmp -s /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.set /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.prev || { cp /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.set /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config; cp /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.set /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.prev; }
export MAKEFLAGS= ; [ -d /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/user_headers ] || make -C /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72 KCFLAGS="-ffile-prefix-map=/home/chitanda/lede/build_dir/target-mipsel_24kc_musl=target-mipsel_24kc_musl" HOSTCFLAGS="-O2 -I/home/chitanda/lede/staging_dir/host/include  -Wall -Wmissing-prototypes -Wstrict-prototypes" CROSS_COMPILE="mipsel-openwrt-linux-musl-" ARCH="mips" KBUILD_HAVE_NLS=no KBUILD_BUILD_USER="" KBUILD_BUILD_HOST="" KBUILD_BUILD_TIMESTAMP="Wed Oct 21 06:52:21 2020" KBUILD_BUILD_VERSION="0" HOST_LOADLIBES="-L/home/chitanda/lede/staging_dir/host/lib" KBUILD_HOSTLDLIBS="-L/home/chitanda/lede/staging_dir/host/lib" CONFIG_SHELL="bash" V=''  cmd_syscalls= KERNELRELEASE=5.4.72 CC="mipsel-openwrt-linux-musl-gcc" INSTALL_HDR_PATH=/home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/user_headers headers_install
grep '=[ym]' /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.config.set | LC_ALL=C sort | mkhash md5 > /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.vermagic
touch /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.configured
rm -f /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/root-ramips/init
make -C /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72 KCFLAGS="-ffile-prefix-map=/home/chitanda/lede/build_dir/target-mipsel_24kc_musl=target-mipsel_24kc_musl" HOSTCFLAGS="-O2 -I/home/chitanda/lede/staging_dir/host/include  -Wall -Wmissing-prototypes -Wstrict-prototypes" CROSS_COMPILE="mipsel-openwrt-linux-musl-" ARCH="mips" KBUILD_HAVE_NLS=no KBUILD_BUILD_USER="" KBUILD_BUILD_HOST="" KBUILD_BUILD_TIMESTAMP="Wed Oct 21 06:52:21 2020" KBUILD_BUILD_VERSION="0" HOST_LOADLIBES="-L/home/chitanda/lede/staging_dir/host/lib" KBUILD_HOSTLDLIBS="-L/home/chitanda/lede/staging_dir/host/lib" CONFIG_SHELL="bash" V=''  cmd_syscalls= KERNELRELEASE=5.4.72 CC="mipsel-openwrt-linux-musl-gcc" all modules
make[5]: Entering directory '/home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72'
  CALL    scripts/checksyscalls.sh
  CALL    scripts/atomic/check-atomics.sh
  CHK     include/generated/compile.h
  GEN     .version
  CHK     include/generated/compile.h
  LD      vmlinux.o
mipsel-openwrt-linux-musl-ld: drivers/net/ethernet/ralink/mtk_offload.o: in function `mtk_offload_check_rx':
target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/drivers/net/ethernet/ralink/mtk_offload.c:508: multiple definition of `mtk_offload_check_rx'; drivers/net/ethernet/mediatek/mtk_offload.o:target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/drivers/net/ethernet/mediatek/mtk_offload.c:122: first defined here
mipsel-openwrt-linux-musl-ld: drivers/net/ethernet/ralink/mtk_debugfs.o: in function `mtk_ppe_debugfs_init':
target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/drivers/net/ethernet/ralink/mtk_debugfs.c:103: multiple definition of `mtk_ppe_debugfs_init'; drivers/net/ethernet/mediatek/mtk_ppe_debugfs.o:target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/drivers/net/ethernet/mediatek/mtk_ppe_debugfs.c:104: first defined here
make[5]: *** [Makefile:1103: vmlinux] Error 1
make[5]: Leaving directory '/home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72'
make[4]: *** [Makefile:26: /home/chitanda/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.image] Error 2
make[4]: Leaving directory '/home/chitanda/lede/target/linux/ramips'
make[3]: *** [Makefile:13: install] Error 2
make[3]: Leaving directory '/home/chitanda/lede/target/linux'
time: target/linux/install#6.42#4.14#10.42
make[2]: *** [target/Makefile:25: target/linux/install] Error 2
make[2]: Leaving directory '/home/chitanda/lede'
make[1]: *** [target/Makefile:19: /home/chitanda/lede/staging_dir/target-mipsel_24kc_musl/stamp/.target_install] Error 2
make[1]: Leaving directory '/home/chitanda/lede'
make: *** [/home/chitanda/lede/include/toplevel.mk:224：world] 错误 2
```

最新commit同样的错误
```
2020-10-21T10:05:40.7795031Z mipsel-openwrt-linux-musl-ld: drivers/net/ethernet/ralink/mtk_offload.o: in function `mtk_offload_check_rx':
2020-10-21T10:05:40.7804506Z target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/drivers/net/ethernet/ralink/mtk_offload.c:508: multiple definition of `mtk_offload_check_rx'; drivers/net/ethernet/mediatek/mtk_offload.o:target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/drivers/net/ethernet/mediatek/mtk_offload.c:122: first defined here
2020-10-21T10:05:40.7813540Z mipsel-openwrt-linux-musl-ld: drivers/net/ethernet/ralink/mtk_debugfs.o: in function `mtk_ppe_debugfs_init':
2020-10-21T10:05:40.7822521Z target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/drivers/net/ethernet/ralink/mtk_debugfs.c:103: multiple definition of `mtk_ppe_debugfs_init'; drivers/net/ethernet/mediatek/mtk_ppe_debugfs.o:target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/drivers/net/ethernet/mediatek/mtk_ppe_debugfs.c:104: first defined here
2020-10-21T10:05:42.0269201Z Makefile:1103: recipe for target 'vmlinux' failed
2020-10-21T10:05:42.0272716Z make[5]: *** [vmlinux] Error 1
2020-10-21T10:05:42.0273852Z make[5]: Leaving directory '/home/runner/work/newifi3/newifi3/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72'
2020-10-21T10:05:42.0275707Z Makefile:25: recipe for target '/home/runner/work/newifi3/newifi3/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.image' failed
2020-10-21T10:05:42.0277036Z make[4]: Leaving directory '/home/runner/work/newifi3/newifi3/lede/target/linux/ramips'
2020-10-21T10:05:42.0278000Z Makefile:13: recipe for target 'install' failed
2020-10-21T10:05:42.0278949Z make[3]: Leaving directory '/home/runner/work/newifi3/newifi3/lede/target/linux'
2020-10-21T10:05:42.0279667Z time: target/linux/install#646.33#65.34#363.03
2020-10-21T10:05:42.0297907Z make[4]: *** [/home/runner/work/newifi3/newifi3/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.image] Error 2
2020-10-21T10:05:42.0298868Z target/Makefile:23: recipe for target 'target/linux/install' failed
2020-10-21T10:05:42.0299704Z make[3]: *** [install] Error 2
2020-10-21T10:05:42.0300472Z make[2]: *** [target/linux/install] Error 2
2020-10-21T10:05:42.0302369Z make[2]: Leaving directory '/home/runner/work/newifi3/newifi3/lede'
2020-10-21T10:05:42.0319885Z target/Makefile:19: recipe for target '/home/runner/work/newifi3/newifi3/lede/staging_dir/target-mipsel_24kc_musl/stamp/.target_install' failed
2020-10-21T10:05:42.0341340Z make[1]: *** [/home/runner/work/newifi3/newifi3/lede/staging_dir/target-mipsel_24kc_musl/stamp/.target_install] Error 2
2020-10-21T10:05:42.0352644Z make[1]: Leaving directory '/home/runner/work/newifi3/newifi3/lede'
2020-10-21T10:05:42.0367910Z Build failed - please re-run with -j1 to see the real error message
2020-10-21T10:05:42.0369940Z /home/runner/work/newifi3/newifi3/lede/include/toplevel.mk:222: recipe for target 'world' failed
2020-10-21T10:05:42.0375229Z make: *** [world] Error 1
2020-10-21T10:05:42.0467809Z ##[error]Process completed with exit code 2.
```

似的，我也有类似的错误。

> 
  AR      arch/mips/math-emu/lib.a
  EXPORTS arch/mips/math-emu/lib-ksyms.o
  AR      arch/mips/math-emu/built-in.a
  AR      virt/lib/built-in.a
  AR      virt/built-in.a
  GEN     .version
  CHK     include/generated/compile.h
  LD      vmlinux.o
mipsel-openwrt-linux-musl-ld: drivers/net/ethernet/ralink/mtk_offload.o: in function `mtk_offload_check_rx':
target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/drivers/net/ethernet/ralink/mtk_offload.c:508: multiple definition of `mtk_offload_check_rx'; drivers/net/ethernet/mediatek/mtk_offload.o:target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/drivers/net/ethernet/mediatek/mtk_offload.c:122: first defined here
mipsel-openwrt-linux-musl-ld: drivers/net/ethernet/ralink/mtk_debugfs.o: in function `mtk_ppe_debugfs_init':
target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/drivers/net/ethernet/ralink/mtk_debugfs.c:103: multiple definition of `mtk_ppe_debugfs_init'; drivers/net/ethernet/mediatek/mtk_ppe_debugfs.o:target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/drivers/net/ethernet/mediatek/mtk_ppe_debugfs.c:104: first defined here
make[5]: *** [Makefile:1103: vmlinux] Error 1
make[5]: Leaving directory '/home/ago/Desktop/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72'
make[4]: *** [Makefile:26: /home/ago/Desktop/lede/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.72/.image] Error 2
make[4]: Leaving directory '/home/ago/Desktop/lede/target/linux/ramips'
make[3]: *** [Makefile:13: install] Error 2
make[3]: Leaving directory '/home/ago/Desktop/lede/target/linux'
time: target/linux/install#659.29#62.61#714.83
make[2]: *** [target/Makefile:25: target/linux/install] Error 2
make[2]: Leaving directory '/home/ago/Desktop/lede'
make[1]: *** [target/Makefile:19: /home/ago/Desktop/lede/staging_dir/target-mipsel_24kc_musl/stamp/.target_install] Error 2
make[1]: Leaving directory '/home/ago/Desktop/lede'
make: *** [/home/ago/Desktop/lede/include/toplevel.mk:224：world] 错误 2

>

--- a/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_debugfs.c
+++ b/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_debugfs.c
@@ -33,7 +33,7 @@ static const char *mtk_foe_packet_type_str[] = {
 #define IPV4_HNAPT                      0
 #define IPV4_HNAT                       1
 #define IS_IPV4_HNAPT(x)	(((x)->bfib1.pkt_type == IPV4_HNAPT) ? 1: 0)
-struct mtk_eth *_eth;
+static struct mtk_eth *_eth;
 #define es(entry)		(mtk_foe_entry_state_str[entry->bfib1.state])
 //#define ei(entry, end)		(MTK_PPE_TBL_SZ - (int)(end - entry))
 #define ei(entry, end)		(MTK_PPE_ENTRY_CNT - (int)(end - entry))
@@ -99,7 +99,7 @@ static const struct file_operations mtk_ppe_debugfs_foe_fops = {
 	.release = single_release,
 };
 
-int mtk_ppe_debugfs_init(struct mtk_eth *eth)
+int ra_ppe_debugfs_init(struct mtk_eth *eth)
 {
 	struct dentry *root;
 

--- a/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c
+++ b/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c
@@ -961,7 +961,7 @@ static int fe_poll_rx(struct napi_struct *napi, int budget,
 					       RX_DMA_VID(trxd.rxd3));
 
 #ifdef CONFIG_NET_RALINK_OFFLOAD
-		if (mtk_offload_check_rx(priv, skb, trxd.rxd4) == 0) {
+		if (ra_offload_check_rx(priv, skb, trxd.rxd4) == 0) {
 #endif
 			stats->rx_packets++;
 			stats->rx_bytes += pktlen;
@@ -1309,7 +1309,7 @@ static int fe_open(struct net_device *dev)
 	fe_int_enable(priv->soc->tx_int | priv->soc->rx_int);
 	netif_start_queue(dev);
 #ifdef CONFIG_NET_RALINK_OFFLOAD
-	mtk_ppe_probe(priv);
+	ra_ppe_probe(priv);
 #endif
 
 	return 0;
@@ -1348,7 +1348,7 @@ static int fe_stop(struct net_device *dev)
 	fe_free_dma(priv);
 
 #ifdef CONFIG_NET_RALINK_OFFLOAD
-	mtk_ppe_remove(priv);
+	ra_ppe_remove(priv);
 #endif
 
 	return 0;

--- a/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.h
+++ b/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.h
@@ -519,14 +519,14 @@ static inline void *priv_netdev(struct fe_priv *priv)
 	return (char *)priv - ALIGN(sizeof(struct net_device), NETDEV_ALIGN);
 }
 
-int mtk_ppe_probe(struct fe_priv *eth);
-void mtk_ppe_remove(struct fe_priv *eth);
+int ra_ppe_probe(struct fe_priv *eth);
+void ra_ppe_remove(struct fe_priv *eth);
 int mtk_flow_offload(struct fe_priv *eth,
 		     enum flow_offload_type type,
 		     struct flow_offload *flow,
 		     struct flow_offload_hw_path *src,
 		     struct flow_offload_hw_path *dest);
-int mtk_offload_check_rx(struct fe_priv *eth, struct sk_buff *skb, u32 rxd4);
+int ra_offload_check_rx(struct fe_priv *eth, struct sk_buff *skb, u32 rxd4);
 
 
 #endif /* FE_ETH_H */

--- a/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_offload.c
+++ b/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_offload.c
@@ -301,7 +301,7 @@ static int mtk_init_foe_table(struct mtk_eth *eth)
 	return 0;
 }
 
-static int mtk_ppe_start(struct mtk_eth *eth)
+static int ra_ppe_start(struct mtk_eth *eth)
 {
 	int ret;
 
@@ -425,7 +425,7 @@ static int mtk_ppe_busy_wait(struct mtk_eth *eth)
 	return -ETIMEDOUT;
 }
 
-static int mtk_ppe_stop(struct mtk_eth *eth)
+static int ra_ppe_stop(struct mtk_eth *eth)
 {
 	u32 r1 = 0, r2 = 0;
 	int i;
@@ -501,7 +501,7 @@ static void mtk_offload_keepalive(struct fe_priv *eth, unsigned int hash)
 	rcu_read_unlock();
 }
 
-int mtk_offload_check_rx(struct fe_priv *eth, struct sk_buff *skb, u32 rxd4)
+int ra_offload_check_rx(struct fe_priv *eth, struct sk_buff *skb, u32 rxd4)
 {
 	unsigned int hash;
 
@@ -519,22 +519,22 @@ int mtk_offload_check_rx(struct fe_priv *eth, struct sk_buff *skb, u32 rxd4)
 	}
 }
 
-int mtk_ppe_probe(struct mtk_eth *eth)
+int ra_ppe_probe(struct mtk_eth *eth)
 {
 	int err;
 
-	err = mtk_ppe_start(eth);
+	err = ra_ppe_start(eth);
 	if (err)
 		return err;
 
-	err = mtk_ppe_debugfs_init(eth);
+	err = ra_ppe_debugfs_init(eth);
 	if (err)
 		return err;
 
 	return 0;
 }
 
-void mtk_ppe_remove(struct mtk_eth *eth)
+void ra_ppe_remove(struct mtk_eth *eth)
 {
-	mtk_ppe_stop(eth);
+	ra_ppe_stop(eth);
 }

--- a/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_offload.h
+++ b/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_offload.h
@@ -255,6 +255,6 @@ enum mtk_foe_cpu_reason {
 #define MTK_PPE_TBL_SZ			\
 			(MTK_PPE_ENTRY_CNT * sizeof(struct mtk_foe_entry))
 
-int mtk_ppe_debugfs_init(struct mtk_eth *eth);
+int ra_ppe_debugfs_init(struct mtk_eth *eth);
 
 

